src/pages/signin/SignInPage.js

```diff
@@ -110,41 +113,41 @@ class SignInPage extends Component {
             (!this.props.account.validated || this.props.account.forgotPassword) &&
             !showUnlinkLoginForm &&
             !Permissions.canUsePasswordlessLogins(this.props.betas);
-
+    
         let welcomeHeader = '';
         let welcomeText = '';
         if (showValidateCodeForm) {
             if (this.props.account.requiresTwoFactorAuth) {
                 // We will only know this after a user signs in successfully, without their 2FA code
-                welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
+                welcomeHeader = this.props.isSmallScreenWidth || this.props.isInModal ? '' : this.props.translate('welcomeText.welcomeBack');
```

Comment:
We can make it a new variable with check for `this.props.isSmallScreenWidth || this.props.isInModal` and use it everywhere here.